tracecat/actions/integrations/itsm/jira.py

```diff
@@ -0,0 +1,148 @@
+)
+async def create_issue(
+    atlassian_domain: Annotated[
+        str,
+        Field(
+            ...,
+            description="Your Atlassian domain. (i.e. https://your-jira-instance.atlassian.net)",
+        ),
+    ],
+    project_id: Annotated[
+        str,
+        Field(
+            ...,
+            description="The Jira Project ID of the project you wish to create an issue in.",
+        ),
+    ],
+    issue_summary: Annotated[str, Field(..., description="The issue summary (title).")],
+    issue_priority: Annotated[
+        str, Field(..., description="The severity of the issue.")
+    ],
+    issue_type: Annotated[
+        str, Field(..., description="The ID of the Issue Type for the issue.")
+    ],
+    optional_fields: Annotated[
+        dict,
+        Field(
+            ...,
+            description="Optional dictionary of fields you wish to add into the issue.",
+        ),
+    ] = None,
+) -> list[dict[str, Any]]:
+    url = f"{atlassian_domain}/rest/api/3/issue"
+    AUTH_TOKEN = HTTPBasicAuth(os.getenv("JIRA_USERNAME"), os.getenv("JIRA_API_TOKEN"))
+    if not AUTH_TOKEN:
+        raise ValueError("Missing JIRA_USERNAME or JIRA_API_TOKEN")
+
+    headers = {
+        "Accept": "application/json",
+        "User-Agent": "Tracecat",
+        "Content-Type": "application/json",
+    }
+
+    issue_data = {
+        "fields": {
+            "project": {"id": project_id},
+            "summary": issue_summary,
+            "priority": {"id": issue_priority},
+            "issuetype": {"id": issue_type},
+        }
+    }
+
+    # If optional fields are provided, merge them into the issue data
+    if optional_fields:
+        issue_data["fields"].update(optional_fields)
+
+    async with httpx.AsyncClient() as client:
+        response = await client.post(
+            url, headers=headers, auth=AUTH_TOKEN, json=issue_data
+        )
+        response.raise_for_status()
+        return response.json()
+
+
+@registry.register(
+    default_title="Update Jira Issue",
+    description="Updates a Jira ticket with the information provided",
+    display_group="Jira",
+    namespace="integrations.jira",
+    secrets=[jira_secret],
+)
+async def update_issue(
+    atlassian_domain: Annotated[
+        str,
+        Field(
+            ...,
+            description="Your Atlassian domain. (i.e. https://your-jira-instance.atlassian.net)",
+        ),
+    ],
+    issue_key: Annotated[
+        str, Field(..., description="The issue key for the issue you wish to update.")
+    ],
+    update_data: Annotated[
+        dict,
+        Field(
+            ...,
+            description="The data you wish to update the issue with. This should be formatted exactly as the custom field is configured.",
+        ),
+    ] = None,
+) -> list[dict[str, Any]]:
+    url = f"{atlassian_domain}/rest/api/3/issue/{issue_key}"
+    AUTH_TOKEN = HTTPBasicAuth(os.getenv("JIRA_USERNAME"), os.getenv("JIRA_API_TOKEN"))
+    if not AUTH_TOKEN:
```

Comment:
```suggestion
```
There's no need to check that JIRA_USERNAME or JIRA_API_TOKEN doesn't exist as we do that already in the secrets manager!